### ISSUE TITLE
OAuth2 functionality without management views for API only apps

### ISSUE THREAD

**Issue by RaoulDijksman** [doorkeeper-gem/doorkeeper#879]
I would love to see a way to use doorkeeper without the management views. I'm expecting I won't be the only one now that Rails 5 is released with API only apps. Would this be a route you are willing to go down?

This might be able to go hand in hand with issue #710

**Comment by tute** [comment]
I'd like to see this, yes. In such a configuration, how would you edit the applications? Thanks!

**Comment by RaoulDijksman** [comment]
It would be nice if it was possible to tell rails what parts of the engine you would like to mount.
Maybe a api only mode like: `use_doorkeeper :api_only` that wouldn't mount the routes nor add assets to the pipeline that belong to views.

Another option would be to split up the project in to 2 gems. The oauth part and a management part.

**Comment by diddeb** [comment]
I tend to agree on this subject. Perhaps splitting the project is too radical, but some way of disabling the admin UI would be appreciated.

**Comment by tute** [comment]
> Perhaps splitting the project is too radical, but some way of disabling the admin UI would be appreciated.

Wholeheartedly agree. The engine shouldn't always inject the routes, according to the `api_mode` Rails option.

**Comment by tute** [comment]
@RaoulDijksman @diddeb can you please give #893 a try? If it works well, I'll release it soon. Thank you!

**Comment by belgoros** [comment]
So if I'm not mistaken, I have to change in my routes as follows:
```
#routes.rb
use_doorkeeper :api_only
```
Right ? What kind of controller should I set as `base_controller` in `doorkeeper.rb`:
```
base_controller 'ApplicationController'
```
takin in account that in Rails API `ApplicationController` is inherited from `ActionController::API`?

**Comment by prcongithub** [comment]
Is this resolved? I am facing a similar issue.

**Comment by nbulaj** [comment]
@prcongithub nope, it is not implemented yet. We have two PRs with implementation for Rails API, but they need some exploration and work.

**Comment by nbulaj** [comment]
@prcongithub @RaoulDijksman merged in ba374aaec7c031a2ae9695e6a5e097551a8d92c0 , can you check it with `master` branch?

**Comment by prcongithub** [comment]
Cool. Will give it a try soon.
